src/Microsoft.Health.Dicom.Core/Extensions/DicomDatasetExtensions.cs

```diff
@@ -158,5 +149,71 @@ public static void AddValueIfNotNull<T>(this DicomDataset dicomDataset, DicomTag
                 dicomDataset.Add(dicomTag, value);
             }
         }
+
+        public static IDictionary<IndexTag, DicomTag> GetDicomTags(this DicomDataset dicomDataset, IEnumerable<IndexTag> indexTags)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(indexTags, nameof(indexTags));
+            IDictionary<IndexTag, DicomTag> result = new Dictionary<IndexTag, DicomTag>();
+            Dictionary<string, IndexTag> privateTags = new Dictionary<string, IndexTag>();
+            foreach (IndexTag indexTag in indexTags)
+            {
+                if (!indexTag.Tag.IsPrivate)
+                {
+                    result.Add(indexTag, indexTag.Tag);
+                }
+                else
+                {
+                    privateTags.Add(indexTag.Tag.GetPath(), indexTag);
+                }
+            }
+
+            if (privateTags.Count != 0)
+            {
+                // IndexTag don't have privateCreator for private tag, need to fill that part from DicomDataset.
+                foreach (DicomItem item in dicomDataset)
+                {
+                    if (item.Tag.IsPrivate)
+                    {
+                        string tagPath = item.Tag.GetPath();
+                        if (privateTags.ContainsKey(tagPath))
+                        {
+                            IndexTag indexTag = privateTags[tagPath];
+                            if (indexTag.VR == item.ValueRepresentation)
+                            {
+                                result.Add(indexTag, item.Tag);
+                            }
+                        }
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        public static IReadOnlyDictionary<IndexTag, object> GetIndexTagValues(this DicomDataset dicomDataset, IEnumerable<IndexTag> indexTags)
```

Comment:
summary and unit tests
1. when contains both standard and private tag, able to get  data